File: libs/language/src/lib/provider/BootstrapProvider.ts
import { NodeProvider } from '../NodeProvider';
import { BlueNode } from '../model';
import { BaseContentNodeProvider } from './BaseContentNodeProvider';

/**
 * A provider that supplies bootstrap nodes
 * This is a simplified version of the Java implementation
 */
export class BootstrapProvider extends NodeProvider {
  public static readonly INSTANCE: BootstrapProvider = new BootstrapProvider();

  private nodeProvider: NodeProvider;

  private constructor() {
    super();
    this.nodeProvider = new BaseContentNodeProvider();
  }

  override fetchByBlueId(blueId: string): BlueNode[] | null {
    return this.nodeProvider.fetchByBlueId(blueId);
  }
}

File: references/java-sdk/src/main/java/blue/language/provider/BootstrapProvider.java
package blue.language.provider;

import blue.language.NodeProvider;
import blue.language.model.Node;

import java.io.IOException;
import java.util.List;

import static blue.language.provider.ClasspathBasedNodeProvider.NO_PREPROCESSING;

public class BootstrapProvider implements NodeProvider {

    public static final BootstrapProvider INSTANCE = new BootstrapProvider();

    private NodeProvider nodeProvider;

    private BootstrapProvider() {
        try {
            ClasspathBasedNodeProvider transformation = new ClasspathBasedNodeProvider(NO_PREPROCESSING, "transformation");
            //ClasspathBasedNodeProvider core = new ClasspathBasedNodeProvider("core");
            this.nodeProvider = new SequentialNodeProvider(transformation);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }

    }

    @Override
    public List<Node> fetchByBlueId(String blueId) {
        return nodeProvider.fetchByBlueId(blueId);
    }

}